restaurant_bill/src/containers/wrapper.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react';
+import calculateBill from '../scripts/calculate'
+
+class Wrapper extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      people: 2,
+      totalBill: 0,
+      extraTip: 0,
+    }
+    this.handleChange = this.handleChange.bind(this);
```

Comment:
I don't know why you need this line, seems unnecessary since handleChange should be bound to `this` already as per your code

Reply:
both posibilities are allowed, https://medium.com/@samichkhachkhi/typeerror-cannot-read-property-setstate-of-undefined-27c14e5b3072
honestly I would not know which is the best one